dpctl/memory/_sycl_usm_array_interface_utils.pxi

```diff
@@ -0,0 +1,187 @@
+        cdef object ary_shape = ary_iface.get('shape', None)
+        cdef object ary_strides = ary_iface.get('strides', None)
+        cdef object ary_syclobj = ary_iface.get('syclobj', None)
+        cdef Py_ssize_t ary_offset = ary_iface.get('offset', 0)
+        cdef int ary_version = ary_iface.get('version', 0)
+        cdef Py_ssize_t arr_data_ptr = 0
+        cdef DPCTLSyclUSMRef memRef = NULL
+        cdef Py_ssize_t itemsize = -1
+        cdef int writeable = -1
+        cdef int nd = -1
+        cdef DPCTLSyclQueueRef QRef = NULL
+        cdef object dt
+        cdef _USMBufferData buf
+        cdef SyclDevice dev
+        cdef SyclContext ctx
+
+        if ary_version != 1:
+            raise ValueError(("__sycl_usm_array_interface__ is malformed:"
+                              " dict('version': {}) is unexpected."
+                              " The only recognized version is 1.").format(ary_version))
+        if not ary_data_tuple or len(ary_data_tuple) != 2:
+            raise ValueError("__sycl_usm_array_interface__ is malformed:"
+                             " 'data' field is required, and must be a tuple"
+                             " (usm_pointer, is_writeable_boolean).")
+        arr_data_ptr = <Py_ssize_t>ary_data_tuple[0]
+        writeable = 1 if ary_data_tuple[1] else 0
+        # Check that memory and syclobj are consistent:
+        # (USM pointer is bound to this sycl context)
+        memRef = <DPCTLSyclUSMRef>arr_data_ptr
+        QRef = get_queue_ref_from_ptr_and_syclobj(memRef, ary_syclobj)
+        if (QRef is NULL):
+            raise ValueError("__sycl_usm_array_interface__ is malformed:"
+                             " 'data' field is not consistent with 'syclobj' field,"
+                             " the pointer {} is not bound to SyclContext derived"
+                             " from dict('syclobj': {}).".format(hex(arr_data_ptr), ary_syclobj))
+        # shape must be present
+        if ary_shape is None or not (isinstance(ary_shape, collections.abc.Sized) and
+              isinstance(ary_shape, collections.abc.Iterable)):
+            raise ValueError("Shape entry is a required element of "
+                             "`__sycl_usm_array_interface__` dictionary")
+        nd = len(ary_shape)
+        try:
+            dt = np.dtype(ary_typestr)
+            if (dt.hasobject or not (np.issubdtype(dt.type, np.integer) or
+                                     np.issubdtype(dt.type, np.inexact))):
+                raise TypeError("Only integer types, floating and complex floating "
+                                "types are supported.")
+            itemsize = <Py_ssize_t> (dt.itemsize)
+        except TypeError as e:
+            raise ValueError("__sycl_usm_array_interface__ is malformed:"
+                             " dict('typestr': {}) is unexpected. ".format(ary_typestr)) from e
+
+        if (ary_strides is None or (isinstance(ary_strides, collections.abc.Sized) and
```

Comment:
long line. Please reformat to <80 chars.

Reply:
Fix this too, here and in other `dpctl/memory` Cython source files.